src/lib/bm_ncp/ncp_config.cpp

```diff
@@ -3,77 +3,116 @@
 #include "device_info.h"
 #include "FreeRTOS.h"
 #include "bm_serial.h"
+#include "bridgeLog.h"
 
 using namespace cfg;
 
 static Configuration *_usr_cfg;
 static Configuration *_sys_cfg;
 static Configuration *_hw_cfg;
 
-// TODO: BCMP downstream passthrough to be done later. 
 
 static Configuration* get_partition(bm_common_config_partition_e partition);
 static uint8_t* alloc_ncp_key_buffer(uint8_t num_keys, const ConfigKey_t* keys, size_t& len);
 
+// These are the callback functions that BCMP req/rep will call when we get a response
+// from the target node.
+bool _cfg_get_bcmp_cb(uint8_t *payload);
+bool _cfg_set_bcmp_cb (uint8_t *payload);
+bool _cfg_status_request_bcmp_cb(uint8_t *payload);
+bool _cfg_key_del_bcmp_cb(uint8_t *payload);
+
 bool ncp_cfg_get_cb(uint64_t node_id, bm_common_config_partition_e partition, size_t key_len, const char* key) {
     bool rval = false;
-    uint8_t * valueBuf = reinterpret_cast<uint8_t*>(pvPortMalloc(MAX_CONFIG_BUFFER_SIZE_BYTES));
-    configASSERT(valueBuf);
+    uint8_t *valueBuf = NULL;
     do {
-        Configuration* p = get_partition(partition);
-        if(!p) {
-            printf("Invalid partition\n.");
-            break;
-        }
-        size_t value_len = MAX_CONFIG_BUFFER_SIZE_BYTES;
-        if(!p->getConfigCbor(key, key_len, valueBuf, value_len)){
-            printf("Failed to get config.\n");
-            break;
-        }
-        if(bm_serial_cfg_value(node_id, partition, value_len, valueBuf) != BM_SERIAL_OK) {
-            printf("Failed to send cfg\n");
-            break;
+        if (node_id == getNodeId() || node_id == 0) {
+            valueBuf = reinterpret_cast<uint8_t*>(pvPortMalloc(MAX_CONFIG_BUFFER_SIZE_BYTES));
+            configASSERT(valueBuf);
+            Configuration* p = get_partition(partition);
+            if(!p) {
+                printf("Invalid partition\n.");
+                break;
+            }
+            size_t value_len = MAX_CONFIG_BUFFER_SIZE_BYTES;
+            if(!p->getConfigCbor(key, key_len, valueBuf, value_len)){
+                printf("Failed to get config.\n");
+                break;
+            }
+            if(bm_serial_cfg_value(node_id, partition, value_len, valueBuf) != BM_SERIAL_OK) {
+                printf("Failed to send cfg\n");
+                break;
+            }
+        } else {
+            err_t err = ERR_OK;
+            if (!bcmp_config_get(node_id, partition, key_len, key, err, _cfg_get_bcmp_cb)) {
+                static char buf[64];
+                snprintf(buf, sizeof(buf), "Failed to issue bcmp cfg get, err: %" PRIi8 "\n", err);
+                BRIDGE_CFG_LOG_PRINT(buf);
+                break;
+            }
         }
         rval = true;
     } while(0);
-    vPortFree(valueBuf);
+    if (valueBuf != NULL) {
+        vPortFree(valueBuf);
+    }
```

Comment:
I edited these such that both bridge cfg requests and other node cfg requests fall into the do-while(0) statement and if they error they break out of the same statement. I also added the checking of `err` and the return value for the `bcmp_cfg_...` requests.